test/e2e/upgrade_ironic_test.go

```diff
@@ -0,0 +1,59 @@
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func upgrade_ironic() {
+	clientSet := targetCluster.GetClientSet()
+	ironicDeployName := "baremetal-operator-ironic"
+	By("Upgrading ironic image based containers")
+	getIronicDeployment := func() (*appsv1.Deployment, error) {
+		return clientSet.AppsV1().Deployments("baremetal-operator-system").Get(ctx, ironicDeployName, metav1.GetOptions{})
+	}
+	deploy, err := getIronicDeployment()
+	Expect(err).To(BeNil())
+	for i := range deploy.Spec.Template.Spec.Containers {
+		switch deploy.Spec.Template.Spec.Containers[i].Name {
+		case
+			"mariadb",
+			"ironic-api",
+			"ironic-dnsmasq",
+			"ironic-conductor",
+			"ironic-log-watch",
+			"ironic-inspector":
+			deploy.Spec.Template.Spec.Containers[i].Image = "quay.io/metal3-io/ironic:latest"
+		}
+	}
+
+	_, err = clientSet.AppsV1().Deployments("baremetal-operator-system").Update(ctx, deploy, metav1.UpdateOptions{})
+	checkError(err)
```

Comment:
Could you use `Expect(err).To(BeNil())` instead of your `checkError` function here?